File: linux/ardium_source/presentation/11_Benchmark/benchmark.go
// Benchmark: Sieve of Eratosthenes

package main

import (
	"fmt"
	"time"
)

func run_sieve(n int) int {
	count := 0
	flags := make([]bool, n)
	for i := range flags {
		flags[i] = true
	}

	for i := 2; i*i < n; i++ {
		if flags[i] {
			for j := i * i; j < n; j += i {
				flags[j] = false
			}
		}
	}

	for i := 2; i < n; i++ {
		if flags[i] {
			count++
		}
	}
	return count
}

func main() {
	n := 100000000
	fmt.Printf("Finding primes up to %d...\n", n)
	start := time.Now()
	c := run_sieve(n)
	elapsed := time.Since(start)
	fmt.Printf("Found: %d\n", c)
	fmt.Printf("Time: %s\n", elapsed)
}

File: linux/ardium_source/presentation/11_Benchmark/benchmark.swift
// Benchmark: Sieve of Eratosthenes

import Foundation

func run_sieve(_ n: Int) -> Int {
    var count = 0
    var flags = [Bool](repeating: true, count: n)
    
    var i = 2
    while i * i < n {
        if flags[i] {
            var j = i * i
            while j < n {
                flags[j] = false
                j += i
            }
        }
        i += 1
    }
    
    for k in 2..<n {
        if flags[k] {
            count += 1
        }
    }
    return count
}

let n = 100000000
print("Finding primes up to \(n)...")
let start = CFAbsoluteTimeGetCurrent()
let c = run_sieve(n)
let diff = CFAbsoluteTimeGetCurrent() - start
print("Found: \(c)")
print("Time: \(diff)s")
